version/distro/distro.go

```diff
@@ -62,9 +65,20 @@ func linuxDistro() Distro {
 	case haveDir("/usr/syno"):
 		return Synology
 	case have("/usr/local/bin/freenas-debug"):
-		// TrueNAS Scale runs on debian
+		// TrueNAS Scale runs on debian so test for it before Debian.
 		return TrueNAS
 	case have("/etc/debian_version"):
+		// Ubuntu also has an /etc/debian_version file, so see if it's actually Ubuntu.
+		isUbuntu := false
+		lineread.File("/etc/os-release", func(line []byte) error {
+			if mem.HasPrefix(mem.B(line), mem.S("ID=ubuntu")) {
```

Comment:
There are a largish number of almost-Ubuntu distributions where the ID=pop or zorin or elementary but ID_LIKE will be "ubuntu"

Depending on what you're planning, a LikeUbuntu() method might accommodate more of these distros.

Reply:
Yeah, at some point I might need a whole family tree set of methods. I'm kinda resisting the urge to do it before I need it, as fun as it sounds.

Reply:
Oooh there could be a `begat` package where Debian begat Ubuntu, and Ubuntu begat PopOS, and...

Reply:
That reminds me of the user agent detection logic we had at my old job. Chrome and Safari shared some bugs/workarounds because they dated to behavior from before the WebKit/Blink fork in 2013, and Opera and Edge switched rendering engines at some point, etc. so we had a "lineage" mechanism to rationalize some of that.